### DISCUSSION TITLE
Password with special character (@) result in wrong parsing of connection string

### DISCUSSION THREAD

**Discussion by erfannariman** [sqlalchemy/sqlalchemy#6568] [Usage Questions] [answered]
Some of our projects start to fail with SQLAlchemy == 1.4.17 when there is an `@` in the password. The same projects run fine on version 1.4.15. 

How to reproduce:

Create login with password credential: `reallysafe@password`. 
And use the following connection string:

```python
    connection_string = "mssql+pyodbc://{}:{}@{}:1433/{}?driver={}".format(
        os.environ.get("SQL_USER"),
        os.environ.get("SQL_PW"),
        os.environ.get("SQL_SERVER"),
        os.environ.get("SQL_DB"),
        "ODBC Driver 17 for SQL Server",
    )
```

This will work on `1.4.15` and fail on `1.4.17`.

Solution is to use: `urllib.parse.quote_plus(os.environ.get("SQL_PW"))`

From what I could see during debugging is the latter version will parse the password as `reallysafe`,

**Comment by CaselIT** [comment, marked as answer]
Hi,

This is due to https://github.com/sqlalchemy/sqlalchemy/issues/6482.

Short version: the previous url regexp was wrong and has now been fixed, as a side effect an `@` in the password is treated differently. As documented special chars in the password should be percent encoded.
